File: 7.stacks_and_queues/lesson_7_nesting.py
def solution(S):
    s = len(S)
    open_brackets = 0
    close_brackets = 0

    for elem in S:
        if(elem == '('):
            open_brackets += 1
        else:
            close_brackets += 1
    
    if(open_brackets != close_brackets):
        return 0

    stack = [''] * s
    top_elem_index = 0

    for index in range(s):
        if(S[index] == '('):
            stack[top_elem_index] = S[index]
            top_elem_index += 1
        else: 
            if(top_elem_index == 0):
                return 0
            top_elem_index -= 1
    return 1

File: 7.stacks_and_queues/lesson_7_nesting.js
function solution(S) {
    let stack = new Array()
    for (let i = 0; i < S.length; i++) {
        switch(S[i]) {
            case ')':
                if(stack.length < 0 || stack.pop() != '(') {
                    return 0
                }
                break
            default:
                stack.push(S[i])
                break
        }
    }
    if(stack.length == 0) {
        return 1
    }
    return 0
}